src/mkfs.fat.c

```diff
@@ -1372,6 +1393,35 @@ static void setup_tables(void)
 	error ("failed whilst writing " errstr);	\
   } while(0)
 
+static void process_bad_blocks(void)
+{
+    printf("%d bad block%s\n", bad_blocks, (bad_blocks > 1) ? "s" : "");
+
+    if (size_fat == 32) {
+	struct fat32_fsinfo *info;
+	uint32_t free_clusters;
+	unsigned int root_cluster;
+
+	/* fsinfo structure is at offset 0x1e0 in info sector by observation */
+	info = (struct fat32_fsinfo *)(info_sector_buffer + 0x1e0);
+
+	/* reduce amount of free clusters accordingly */
+	free_clusters = le32toh(info->free_clusters);
+	if (free_clusters < bad_clusters)
+		die("too many bad clusters");
+	info->free_clusters = htole32(free_clusters - bad_clusters);
+
+	/* verify that root directory is intact */
+	root_cluster = 2;
```

Comment:
I would suggest to read this value (as it is already filled) instead harcoding constant 2 on another place.
```suggestion
	root_cluster = le32toh(bs.fat32.root_cluster);
```

Reply:
Oh right, that makes sense. Applied and thank you for the reviews!